okapi-core/src/main/java/org/folio/okapi/managers/InternalModule.java

```diff
@@ -1056,6 +1068,19 @@ public static ModuleDescriptor moduleDescriptor(String okapiVersion) {
     }
   }
 
+  private Future<Void> importModules(ProxyContext pc, String body) {
+    try {
+      HttpServerRequest req = pc.getCtx().request();
+      final boolean check = ModuleUtil.getParamBoolean(req, "check", true);
+      final boolean preRelease = ModuleUtil.getParamBoolean(req, "preRelease", true);
+      final boolean npmSnapshot = ModuleUtil.getParamBoolean(req, "npmSnapshot", true);
+      final ModuleDescriptor[] modules = Json.decodeValue(body, ModuleDescriptor[].class);
+      return moduleManager.createList(Arrays.asList(modules), check, preRelease, npmSnapshot);
```

Comment:
The each module descriptor should be validated the same way as createModule validates it.

Reply:
ok